src/modules/shortcut.ts

```diff
@@ -0,0 +1,202 @@
+import {
+	command,
+	default as CookiecordClient,
+	Module,
+	listener,
+} from 'cookiecord';
+import { Message, MessageEmbed, User } from 'discord.js';
+import { send } from 'process';
+import { BaseEntity, DeepPartial, ObjectType } from 'typeorm';
+import { Shortcut } from '../entities/Shortcut';
+import { BLOCKQUOTE_GREY, BLOCKQUOTE_GREY_NUMBER, TS_BLUE } from '../env';
+import { sendWithMessageOwnership } from '../util/send';
+
+// https://stackoverflow.com/a/3809435
+const LINK_REGEX = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/;
+
+export class ShortcutModule extends Module {
+	constructor(client: CookiecordClient) {
+		super(client);
+	}
+
+	@listener({ event: 'message' })
+	async runShortcut(msg: Message) {
+		const { content } = msg;
+		const commandPart = content.split(' ')[0];
+		if (!commandPart.startsWith('!')) return;
+		const id = commandPart.slice(1);
+		if (this.client.commandManager.getByTrigger(id)) return;
+		const shortcut = await this.getShortcut(id);
+		if (!shortcut) return;
+		if (shortcut.content)
+			return await sendWithMessageOwnership(msg, shortcut.content);
+		const owner = await this.client.users.fetch(shortcut.owner);
+		const embed = new MessageEmbed({
+			...shortcut,
+			image: undefined,
+		});
+		embed.setAuthor(owner.tag, owner.displayAvatarURL());
+		if (shortcut.image) embed.setImage(shortcut.image);
+		await sendWithMessageOwnership(msg, { embed });
+		await Shortcut.createQueryBuilder()
+			.update()
+			.set({ uses: () => 'uses + 1' })
+			.execute();
+	}
+
+	@command({
+		single: true,
+		description: 'Shortcut: Create or edit a shortcut',
```

Comment:
You can remove the `Shortcut:` prefix here and in a few other similliar instances.

Reply:
Those are used by the `!help` generator; if missing they'll be grouped under `Misc`.

Reply:
Ah cool, I guess that's new.